src/estimators/bloom_filters.py

```diff
@@ -419,6 +421,53 @@ def __call__(self, sketch_list):
         union, self._weights)]
 
 
+class FirstMomentGlobalNoiseEstimator(EstimatorBase):
+  """This estimator unions the input sketches, generate geometric noise,
+     appends that many fake buckets to the sketch union and calls the
+     FirstMomentEstimator.
+  """
+
+  # TODO(kungfucraig): Figure out how to get the framework to pass
+  # a random state.
+  def __init__(self, epsilon, method, random_state=None):
+    self.epsilon = epsilon
+    assert method == FirstMomentEstimator.METHOD_EXP, "only METHOD_EXP is supported."
+    self.estimator = FirstMomentEstimator(method=method)
+    self.random_state = random_state
+
+  def __call__(self, sketch_list):
+    if not sketch_list:
+      return 0
+    union = sketch_list[0]
+    for cur_sketch in sketch_list[1:]:
+      union.assert_compatible(cur_sketch)
+      union.sketch = 1 - (1 - union.sketch) * (1 - cur_sketch.sketch)
+
+    noiser = GeometricEstimateNoiser(
+        self.epsilon, random_state=self.random_state)
+    noise = noiser(0.)
+
+    a = union.decay_rate
```

Comment:
This we agree to this approach I'll do some refactoring to pull out the code that this shares with the FirstMomentEstimator, or make some of the FirstMomentEstimator's methods public.